lib/ansiblelint/utils.py

```diff
@@ -742,3 +750,105 @@ def normpath(path):
     make this user configurable.
     """
     return os.path.relpath(path)
+
+
+def is_playbook(filename):
+    """
+    Check if the file is a playbook.
+
+    Given a filename, it should return true if it looks like a playbook. The
+    function is not supposed to raise exceptions.
+    """
+    # we assume is a playbook if we loaded a sequence of dictionaries where
+    # at least one of these keys is present:
+    playbooks_keys = {
+        "gather_facts",
+        "hosts",
+        "import_playbook",
+        "post_tasks",
+        "pre_tasks",
+        "roles"
+        "tasks",
+    }
+
+    # makes it work with Path objects by converting them to strings
+    if not isinstance(filename, six.string_types):
+        filename = str(filename)
+
+    try:
+        f = parse_yaml_from_file(filename)
+    except Exception as e:
+        print(
+            "Warning: Failed to load %s with %s, assuming is not a playbook."
+            % (filename, e))
+    else:
+        if (
+            isinstance(f, AnsibleSequence)
+            and playbooks_keys.intersection(next(iter(f), {}).keys())
+        ):
+            return True
+    return False
+
+
+def get_playbooks_and_roles(options={}):
```

Comment:
Mutable default values are harmful
```suggestion
def get_playbooks_and_roles(options=None):
    if options is None:
        options = {}

```